src/stream/node/mod.rs

```diff
@@ -237,29 +232,41 @@ where
             self.node_state.message_buffer().append(message.raw())?;
             return Ok(());
         }
+
+        #[cfg(feature = "metrics")]
+        self.node_metrics.start_timer();
```

Comment:
As this measurement is just a local concern, there is no need to store it in NodeMetrics. Use ``std::time::Instant`` like below.  The only thing is turning the elapsed time into an f64 as the histogram requires it I think?  there is a ``as_secs_f64`` function on Duration, but we might want it in something like nanoseconds/milliseconds.

```rust
#[cfg(feature = "metrics")]
let start = Instant::now();

match message {
   MessageContainer::Raw(r) => self.handle_events(r.sender, r.events),
   MessageContainer::Local(l) => self.handle_events(l.sender, l.events),
}

#[cfg(feature = "metrics")]
{
  let elapsed = start.elapsed(); //  needs to be converted to f64 for histogram?
  histogram!("....", elapsed);
}
```